src/indexmap.rs

```diff
@@ -0,0 +1,2539 @@
+#![allow(unsafe_code)]
```

Comment:
This deserves a comment explaining why we're doing this.

Reply:
Would something along the lines of 'this is vendored well-tested code that we didn't want to modify' be sufficient? Further than that, I'll need some assistance with this comment.

Reply:
Yep, that's exactly the sort of thing I was hoping for.